src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -508,10 +511,14 @@ public FormValidation doCheckProjectName(
                 }
             else if (value.indexOf('$') >= 0)
                 result = FormValidation.warning(Messages.CopyArtifact_ParameterizedName());
-            else
+            else {
+                AbstractProject nearest = AbstractProject.findNearest(value);
+                String alternative = nearest != null ? nearest.getRelativeNameFrom(parent) : "?";
```

Comment:
+1 for `Items.findNearest`.

This change is supposed to avoid NPE when there are only `WorkflowJob`s and no `AbstractProject`s.
`Items.findNearest` enables us to display also a `WorkflowJob` as a candidate.